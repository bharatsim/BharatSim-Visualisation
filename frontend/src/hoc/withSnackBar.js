import React from 'react';
import { SnackbarProvider } from 'notistack';
import Button from '@material-ui/core/Button';
import { makeStyles } from '@material-ui/core/styles';
import failureIcon from '../assets/images/error-icon.svg';
import successIcon from '../assets/images/success-icon.svg';

const useSnackBarStyles = makeStyles((theme) => ({
  contentRoot: {
    padding: theme.spacing(1, 4),
  },
  message: {
    padding: 0,
  },
  variantSuccess: {
    backgroundColor: `#EAF4E5 !important`,
    borderColor: `${theme.palette.success.main}33`,
    border: '1px solid',
    color: `${theme.palette.success.dark} !important`,
    ...theme.typography.body2,
    borderRadius: theme.spacing(1),
    boxShadow: 'unset',
    opacity: 1,
    minWidth: theme.spacing(120),
  },
  variantError: {
    backgroundColor: `#FBE9E4 !important`,
    borderColor: `${theme.palette.error.light}33`,
    border: '1px solid',
    color: `${theme.palette.error.dark} !important`,
    ...theme.typography.body2,
    borderRadius: theme.spacing(1),
    boxShadow: 'unset',
    opacity: 1,
    minWidth: theme.spacing(120),
  },
}));

const useIconStyles = makeStyles((theme) => ({
  iconRoot: {
    marginRight: theme.spacing(2),
    height: theme.spacing(10),
    width: theme.spacing(10),
  },
}));

function withSnackBar(WrappedComponent) {
  return (props) => {
    const snackBarClasses = useSnackBarStyles();
    const iconClasses = useIconStyles();
    const snackBarRef = React.createRef();
    const onClickDismiss = (key) => () => {
      snackBarRef.current.closeSnackbar(key);
    };
    return (
      <SnackbarProvider
        maxSnack={3}
        ref={snackBarRef}
        anchorOrigin={{ horizontal: 'right', vertical: 'bottom' }}
        classes={snackBarClasses}
        iconVariant={{
          error: <img src={failureIcon} alt="error" className={iconClasses.iconRoot} />,
          success: <img src={successIcon} alt="success" className={iconClasses.iconRoot} />,
        }}
        action={(key) => <Button onClick={onClickDismiss(key)}>Dismiss</Button>}
      >
        <WrappedComponent {...props} />
      </SnackbarProvider>
    );
  };
}

export default withSnackBar;
