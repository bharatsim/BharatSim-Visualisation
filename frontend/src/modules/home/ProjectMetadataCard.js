import React from 'react';
import PropTypes from 'prop-types';
import { Box, Typography } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';

const styles = makeStyles((theme) => ({
  cardContainer: {
    padding: theme.spacing(4, 6),
    height: theme.spacing(32),
    textTransform: 'capitalize',
    boxShadow:
      '0px 1px 5px rgba(0, 0, 0, 0.2), 0px 3px 4px rgba(0, 0, 0, 0.12), 0px 2px 4px rgba(0, 0, 0, 0.14)',
    borderRadius: theme.spacing(1),
    '&:hover': {
      boxShadow:
        '0px 3px 5px rgba(0, 0, 0, 0.2), 0px 1px 18px rgba(0, 0, 0, 0.12), 0px 6px 10px rgba(0, 0, 0, 0.14)',
      cursor: 'pointer',
    },
  },
}));

function ProjectMetadataCard({ project, onProjectClick }) {
  const { name, _id } = project;
  const classes = styles();
  return (
    <Box className={classes.cardContainer} onClick={() => onProjectClick(_id)}>
      <Typography variant="subtitle2">{name}</Typography>
    </Box>
  );
}

ProjectMetadataCard.propTypes = {
  project: PropTypes.shape({
    name: PropTypes.string,
    _id: PropTypes.string,
  }).isRequired,
  onProjectClick: PropTypes.func.isRequired,
};

export default ProjectMetadataCard;
