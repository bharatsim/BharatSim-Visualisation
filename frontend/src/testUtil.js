// eslint-disable-next-line import/no-extraneous-dependencies
import { fireEvent, within } from '@testing-library/react';

export function selectDropDownOption(container, dropDownId, optionId) {
  const dropDown = container.getByTestId(dropDownId);
  fireEvent.mouseDown(within(dropDown).getByRole('button'));
  const options = within(
    within(document.getElementById(`menu-${dropDownId}`)).getByRole('listbox'),
  );
  fireEvent.click(options.getByTestId(`${dropDownId}-${optionId}`));
}
