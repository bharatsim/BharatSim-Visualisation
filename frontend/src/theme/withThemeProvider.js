import React from 'react';
import { MuiThemeProvider, CssBaseline } from '@material-ui/core';

import theme from './theme';

function withThemeProvider(WrappedComponent) {
  return function WithTheme(props) {
    return (
      <MuiThemeProvider theme={theme}>
        <CssBaseline />
        <WrappedComponent {...props} />
      </MuiThemeProvider>
    );
  };
}

export default withThemeProvider;
