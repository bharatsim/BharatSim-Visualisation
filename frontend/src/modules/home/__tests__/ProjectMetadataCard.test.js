import React from 'react';
import { renderWithRedux as render } from '../../../testUtil';
import ProjectMetadataCard from '../ProjectMetadataCard';
import withThemeProvider from '../../../theme/withThemeProvider';
import withSnackBar from '../../../hoc/snackbar/withSnackBar';

describe('ProjectMetadataCard', () => {
  const ProjectMetadataCardWithProviders = withThemeProvider(withSnackBar(ProjectMetadataCard));
  it('should match snapshot ProjectMetadataCard', async () => {
    const { container } = render(
      <ProjectMetadataCardWithProviders
        project={{ _id: 'projectId', name: 'projectName' }}
        onProjectClick={jest.fn()}
        deleteProject={jest.fn()}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
