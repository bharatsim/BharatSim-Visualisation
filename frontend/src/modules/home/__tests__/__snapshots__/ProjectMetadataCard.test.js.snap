// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectMetadataCard should match snapshot ProjectMetadataCard 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-8"
  >
    <div
      class="MuiBox-root MuiBox-root-9 makeStyles-cardContainer-7"
    >
      <div
        class="MuiBox-root MuiBox-root-10 makeStyles-titleContainer-6"
      >
        <h6
          class="MuiTypography-root MuiTypography-subtitle2"
        >
          projectName
        </h6>
        <button
          class="MuiButtonBase-root MuiIconButton-root"
          data-testid="project-menu"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
